ramanujan/enumerators/RelativeGCFEnumerator.py

```diff
@@ -0,0 +1,283 @@
+            if q != 0:  # safety check
+                computed_values.append(trunc_division(precision_factor * p, q))
+                items_computed += 1
+            else:
+                raise ZeroDivisionError()
+
+            if items_computed >= 2:
+                if computed_values[-1] == computed_values[-2]:
+                    return computed_values[-1], result_precision
+
+                if items_computed >= 3:
+                    # if the GCF oscillates between two sub-series - we expect the distance between the two to 
+                    # monotonically decrease. If the GCF doesn't oscillate, we expect the distance between two following
+                    # values to monotonically decrease as well. If none of those apply, we halt the calculation here.
+                    if abs(computed_values[-2] - computed_values[-1]) > abs(computed_values[-3] - computed_values[-2]):
+                        raise NotConverging("Not converging")
+
+    # The locations we calculate the GCF for and the size of the iterator is unrelated. Its likely that if we got here,
+    # that last values of the GCFs p and q were calculated but never got divided and added to computed_values
+    if i != next_gcf_calculation:
+        computed_values.append(trunc_division(precision_factor * p, q))
+        items_computed += 1
+
+    # we'll take the last two calculations, and check for matching digits.
+    # Once the two doesn't match, we'll know that we cannot trust the following digits.
+    res = ''
+    for i, (c1, c2) in enumerate(zip(str(computed_values[-2]), str(computed_values[-1]))):
+        if c1 != c2:
+            break
+        res += c1
+
+    res += '0' * (len(str(computed_values[-1]))-i)
+    return int(res), i
+        
+
+class RelativeGCFEnumerator(AbstractGCFEnumerator):
+    """
+        This enumerator calculates the GCF to an arbitrary dept, until getting to a stable value within the precession
+        required bounds.
+        Useful for GCF that converges slowly.
+    """
+
+    def __init__(self, *args, **kwargs):
+        print('using relative enumerator')
+        super().__init__(*args, **kwargs)
+
+    def _iter_domains_with_cache(self, max_iters):
+        """
+        Yields an and bn pairs in the given domain while keeping cache for one of the series (an or bn)
+        to avoid double calculations.
+
+        We allow calculations up to max_iters, and we'll avoid calculating items we don't use, so the 
+        item in an or bn is only calculated when it's first yielded and then cached.
+        """
+        size_a = self.poly_domains.an_length
+        size_b = self.poly_domains.bn_length
+
+        an_series_iter, bn_series_iter = self.poly_domains.get_calculation_method()
+
+        # The series on the outer loop is only used on one iteration of the out loop. So
+        # we'll cache only the current series for it. All of the domain for the inner series is cached
```

Comment:
"All of the domain for the inner series is cached"
->
"All of the domains* for the inner series are* cached"